Source/MLX/MLXArray+Indexing.swift

```diff
@@ -158,212 +153,46 @@ extension MLXArray {
 
             precondition(dtype == newValue.dtype, "\(dtype) != \(newValue.dtype)")
 
-            // trim off any singleton dimensions
-            let updateShape = Array(newValue.shape.trimmingPrefix { $0 == 1 })
-
             // this is the shape we will set via broadcast.
             let axis = MLX.resolve(axis: axis, ndim: ndim)
             var broadcastShape = self.shape
             broadcastShape[axis] = 1
 
-            let expanded = newValue.reshaped(updateShape).broadcast(to: broadcastShape.asInt32)
+            let expanded =
+                newValue
+                .leadingSingletonDimensionsRemoved(stream: stream)
+                .broadcast(to: broadcastShape.asInt32)
```

Comment:
Some of the implementations of these older deprecated functions are updated to use new functions introduced or to place it on top of the new API.